mvpoly/tests/prime.rs

```diff
@@ -400,3 +400,56 @@ fn test_evaluation_predefined_polynomial() {
     let evaluation = p.eval(&random_evaluation);
     assert_eq!(evaluation, exp_eval);
 }
+
+#[test]
+fn test_eval_pbt_add() {
+    let mut rng = o1_utils::tests::make_test_rng(None);
+
+    let random_evaluation: [Fp; 6] = std::array::from_fn(|_| Fp::rand(&mut rng));
+    let p1 = Dense::<Fp, 6, 4>::random(&mut rng);
+    let p2 = Dense::<Fp, 6, 4>::random(&mut rng);
+    let p3 = p1.clone() + p2.clone();
+    let eval_p1 = p1.eval(&random_evaluation);
+    let eval_p2 = p2.eval(&random_evaluation);
+    let eval_p3 = p3.eval(&random_evaluation);
+    assert_eq!(eval_p3, eval_p1 + eval_p2);
+}
+
+#[test]
+fn test_eval_pbt_sub() {
+    let mut rng = o1_utils::tests::make_test_rng(None);
+
+    let random_evaluation: [Fp; 6] = std::array::from_fn(|_| Fp::rand(&mut rng));
+    let p1 = Dense::<Fp, 6, 4>::random(&mut rng);
+    let p2 = Dense::<Fp, 6, 4>::random(&mut rng);
+    let p3 = p1.clone() - p2.clone();
+    let eval_p1 = p1.eval(&random_evaluation);
+    let eval_p2 = p2.eval(&random_evaluation);
+    let eval_p3 = p3.eval(&random_evaluation);
+    assert_eq!(eval_p3, eval_p1 - eval_p2);
+}
+
+#[test]
+fn test_eval_pbt_mul_by_scalar() {
+    let mut rng = o1_utils::tests::make_test_rng(None);
+
+    let random_evaluation: [Fp; 6] = std::array::from_fn(|_| Fp::rand(&mut rng));
+    let p1 = Dense::<Fp, 6, 4>::random(&mut rng);
+    let c = Fp::rand(&mut rng);
+    let p2 = p1.clone() * Dense::<Fp, 6, 4>::from(c);
+    let eval_p1 = p1.eval(&random_evaluation);
+    let eval_p2 = p2.eval(&random_evaluation);
+    assert_eq!(eval_p2, eval_p1 * c);
+}
+
+#[test]
+fn test_eval_pbt_neg() {
+    let mut rng = o1_utils::tests::make_test_rng(None);
+
+    let random_evaluation: [Fp; 6] = std::array::from_fn(|_| Fp::rand(&mut rng));
+    let p1 = Dense::<Fp, 6, 4>::random(&mut rng);
+    let p2 = -p1.clone();
+    let eval_p1 = p1.eval(&random_evaluation);
+    let eval_p2 = p2.eval(&random_evaluation);
+    assert_eq!(eval_p2, -eval_p1);
+}
```

Comment:
you don't check the multiplication?

Reply:
Addressed in https://github.com/o1-labs/proof-systems/pull/2511/